### ISSUE TITLE
Self-documentation support

### ISSUE THREAD

**Issue by vitaly-t** [vitaly-t/excellent#14]
**This is a possible feature under consideration. Please vote, if you like it.**

For controllers, modules and services, to support optional property `$about`  - object with all details about the entity, so that method [ERoot.analyze] could report all available details about the entity. And it can be available inside controllers also, to log and detect any version-related issues.

The library would not care what the object contains, it would just report its complete content.

**Example:**

```js
// Especially for controllers, you would declare the object outside,
// as there is no point creating it inside each controller instance.

var aboutThisModule = {
       /* NOTE: we do not care what is in here */
        version: '1.2.3',
        description: 'Description for this module',
        comments: 'It is a Beta',
        author: 'John Dow',
        email: 'john.dow@gmail.com',
        lastChanged: '2018-08-12',
        copyright: 'All right reserved, can\'t touch this!'
};

app.addModule('mod', scope => {
    scope.$about = aboutThisModule;
});
```

Perhaps the only question is to whether create a read-only object on the scope, and allow only setting its random properties, or not create it at all, keeping fully optional.

[ERoot.analyze]:https://vitaly-t.github.io/excellent/ERoot.html#analyze

**Comment by vitaly-t** [comment]
After some thought, I decided to implement this only for modules and services, because controllers do not really need such feature.

**Comment by vitaly-t** [comment]
Implemented in version [0.8.5](https://github.com/vitaly-t/excellent/releases/tag/0.8.5), for services and modules.

**Comment by vitaly-t** [comment]
This entire feature was later removed in version [0.8.6] as obsolete, because the statistics now exposes everything live, so no more need to support any special properties.

[0.8.6]:https://github.com/vitaly-t/excellent/releases/tag/0.8.6
